std/algorithm/comparison.d

```diff
@@ -967,24 +968,25 @@ template equal(alias pred = "a == b")
 }
 
 ///
-@safe unittest
+@safe @nogc unittest
 {
     import std.algorithm.comparison : equal;
     import std.math : approxEqual;
+    import std.array : staticArray;
 
-    int[] a = [ 1, 2, 4, 3 ];
-    assert(!equal(a, a[1..$]));
-    assert(equal(a, a));
-    assert(equal!((a, b) => a == b)(a, a));
+    auto a = [ 1, 2, 4, 3 ].staticArray;
```

Comment:
You could also declare those variables as static arrays (which looks better IMHO)
```suggestion
    int[4] a = [ 1, 2, 4, 3 ];
```